dashboard/app/models/user_level.rb

```diff
@@ -26,6 +26,7 @@ class UserLevel < ActiveRecord::Base
   belongs_to :user
   belongs_to :level
   belongs_to :script
+  belongs_to :level_source
```

Comment:
maybe add a comment that UserLevels before #8423 don't have this populated and need to use the Activity/last_attempt logic until a backfill is done (is that right @joshlory ?). I think you are fine for the purposes of the functionality here because it's newer than that.